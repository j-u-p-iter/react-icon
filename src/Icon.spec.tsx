import * as React from "react";
import { cleanup, fireEvent, render } from "react-testing-library";

import { Icon, IconProps } from ".";

describe("Icon", () => {
  let getIconEl: any;

  beforeAll(() => {
    getIconEl = (props: IconProps) => (
      <Icon
        name="some-name"
        svgSpritePath="some-path"
        className="some-class-name"
        data-testid="icon"
        {...props}
      />
    );
  });

  afterEach(cleanup);

  it("renders correctly by default", () => {
    const { container } = render(getIconEl());

    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with all props", () => {
    const { container } = render(getIconEl({ onClick: () => {} }));

    expect(container.firstChild).toMatchSnapshot();
  });

  it("handles click on Icon correctly", () => {
    const onClick = jest.fn();

    const { getByTestId } = render(getIconEl({ onClick }));

    fireEvent.click(getByTestId("icon"));

    expect(onClick).toHaveBeenCalledTimes(1);
    expect(onClick.mock.calls[0][0].currentTarget).toBeDefined();
  });
});
