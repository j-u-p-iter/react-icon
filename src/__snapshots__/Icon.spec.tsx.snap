// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders correctly by default 1`] = `
<svg
  class="some-class-name"
  data-testid="icon"
  name="some-name"
  role="img"
  viewBox="0 0 16 16"
>
  <use
    xlink:href="some-path#some-name"
  />
</svg>
`;

exports[`Icon renders correctly with all props 1`] = `
<svg
  class="some-class-name"
  data-testid="icon"
  name="some-name"
  role="img"
  viewBox="0 0 16 16"
>
  <use
    xlink:href="some-path#some-name"
  />
</svg>
`;
